pkg/webhooks/sgx/sgx.go

```diff
@@ -40,9 +40,39 @@ const (
 	encl                = namespace + "/enclave"
 	epc                 = namespace + "/epc"
 	provision           = namespace + "/provision"
-	provisionAnnotation = namespace + "/needs-provision"
+	quoteProvAnnotation = namespace + "/quote-provider"
+	aesmdQuoteProvKey   = "aesmd"
 )
 
+func getAesmdVolume(needsAesmd bool, epcUserCount int32, aesmdPresent bool) *corev1.Volume {
+	switch {
+	case !needsAesmd:
+		return nil
+	case epcUserCount == 0:
+		return nil
+	case aesmdPresent && epcUserCount >= 2:
+		return &corev1.Volume{
+			Name: "aesmd-socket",
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{
+					Medium: corev1.StorageMediumMemory,
+				},
+			},
+		}
+	default:
```

Comment:
So we get to the default branch when `epcUserCount == 1` and also when `needsAesmd && !aesmdPresent`? Should the latter be an error case? And if there's only one epcuser that is aesmd, why bother with the mounts?

Reply:
all aesmd non-sidecar deployments need the host volume mount: aesmd itself needs it to be able to create the socket and normal containers wanting to use aesmd for quote generation need to be able to access the socket.

Reply:
@ipuustin added comments to explain this